src/modeldata.jl

```diff
@@ -75,9 +75,48 @@ TransformerWeights(p::Config) = TransformerWeights(;
     rms_final_weight = zeros(Float32, p.dim),
     freq_cis_real = zeros(Float32, (p.dim ÷ p.n_heads) ÷ 2, p.seq_len),
     freq_cis_imag = zeros(Float32, (p.dim ÷ p.n_heads) ÷ 2, p.seq_len),
-    wcls = zeros(Float32, p.dim, p.vocab_size) 
+    wcls = zeros(Float32, p.dim, p.vocab_size)
 )
 
+@views function mmap_weights(p::Config, checkpoint_filename)
+    v = Mmap.mmap(checkpoint_filename)
+    # Drop first 7*4=28 bytes that correspond to Config
+    f = reinterpret(Float32, v)[8:end]
```

Comment:
I'm not familiar with the Mmap interface, but the documentation says you can [specify an offset](https://docs.julialang.org/en/v1/stdlib/Mmap/#Mmap.mmap) to skip the header, which would be cleaner if correct.

Reply:
I ended up not doing that (although it worked) and instead just reading out the config here, so I could totally separate the two code paths instead of entangling them.

This also showed me I could mmap directly as `Vector{Float32}` instead of `reinterpret`ing after. That seemed to speed up the `mmap` version all the way up to the same speed as the non-mmap version! So that is nice. Could even be worth `mmap=true` as default.

Reply:
I bet you were triggering some sort of expensive copying behind the scenes